src/ticks/debugger.c

```diff
@@ -1215,33 +1235,26 @@ static int cmd_watch(int argc, char **argv)
         int value = parse_address(argv[2], &corrected_source);
 
         if ( value != -1 ) {
-            elem = malloc(sizeof(*elem));
+            elem = calloc(1, sizeof(*elem));
+            elem->number = next_breakpoint_number++;
             elem->type = breakwrite ? BREAK_WRITE : BREAK_READ;
             elem->value = value;
             elem->enabled = 1;
             LL_APPEND(watchpoints, elem);
-            printf("Adding %s watchpoint at '%s' $%04x (%s)\n",breakwrite ? "write" : "read", corrected_source, value,  resolve_to_label(value));
+            bk.console("Adding %s watchpoint at '%s' $%04x (%s)\n",breakwrite ? "write" : "read", corrected_source, value,  resolve_to_label(value));
         } else {
-            printf("Cannot set watchpoint on '%s'\n", corrected_source);
+            bk.console("Cannot set watchpoint on '%s'\n", corrected_source);
         }
     } else if ( argc == 3 && strcmp(argv[1],"delete") == 0 ) {
         int num = atoi(argv[2]);
-        breakpoint *elem;
-        LL_FOREACH(watchpoints, elem) {
-            num--;
-            if ( num == 0 ) {
-                printf("Deleting watchpoint %d\n",atoi(argv[2]));
-                delete_breakpoint_by_number(elem);
-                break;
-            }
-        }
+        delete_breakpoint_by_number_and_log(num);
```

Comment:
I think this will end up deleting a breakpoint rather than the watch point?

Reply:
Thanks, resolved.